python/cudnn/csa/compressor/api.py

```
@@ -0,0 +1,580 @@
+
+    Mirrors torch's deterministic-mode semantics: strict mode raises, warn-only mode
+    warns and runs.
+    """
+    if torch.are_deterministic_algorithms_enabled():
+        message = (
+            "CSA compressor backward accumulates dAPE with fp32 atomics and is not "
+            "deterministic; torch.use_deterministic_algorithms(True) is set. Use a "
+            "deterministic (eager) implementation instead."
+        )
+        if torch.is_deterministic_algorithms_warn_only_enabled():
+            warnings.warn(message, RuntimeWarning)
```

Ruff (0.15.21)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)